botfront/imports/api/graphql/story/mongo/stories.js

```diff
@@ -42,3 +42,40 @@ export const searchStories = async (projectId, language, search) => {
     ).fetch();
     return matched;
 };
+
+const replaceLine = (story, lineToReplace, newLine) => {
+    // regexp: [ ] = space; + = any number of the characters in the []; $ = end of string
+    const regex = new RegExp(`- ${lineToReplace}([ ]+\n|\n|[ ]+$|$)`, 'g');
+    return story.replace(regex, `- ${newLine}\n`);
+};
+const replaceIndexLine = (index, lineToReplace, newLine) => {
+    const regex = new RegExp(`(^|\n|[ ])${lineToReplace}(\n|$|[ ])`, 'g');
+    return index.replace(regex, `\n${newLine}\n`);
+};
+
+const traverseReplaceLine = (story, lineToReplace, newLine) => {
+    const updatedStory = story;
+    updatedStory.story = replaceLine(updatedStory.story, lineToReplace, newLine);
+    updatedStory.textIndex.contents = replaceIndexLine(updatedStory.textIndex.contents, lineToReplace, newLine);
```

Comment:
Are we sure the `updatedStory.textIndex.contents` exist ? as it could cause an issue if it's undefined